lib/usdUfe/ufe/UsdUIInfoHandler.cpp

```diff
@@ -184,6 +190,38 @@ Ufe::UIInfoHandler::Icon UsdUIInfoHandler::treeViewIcon(const Ufe::SceneItem::Pt
         }
     }
 
+    // Naming convention for third party shader outliner icons:
+    //
+    //  We take the info:id of the shader and make it safe by replacing : with _.
+    //  Then we search the Maya icon paths for a PNG file with that name. If found we will use it.
+    //  Please note that files with _150 and _200 can also be provided for high DPI displays.
+    //
+    //   For example an info:id of:
+    //       MyRenderer:nifty_surface
+    //   Will have this code search the full Maya icon path for a file named:
+    //       MyRenderer_nifty_surface.png
+    //   And will use it if found. At resolution 200%, the file:
+    //       MyRenderer_nifty_surface_200.png
+    //   Will alternatively be used if found.
+    //
+    if (auto shaderSchema = PXR_NS::UsdShadeShader(usdItem->prim())) {
+        static const auto iconContext = PXR_NS::ArDefaultResolverContext(
+            PXR_NS::TfStringSplit(PXR_NS::TfGetenv("XBMLANGPATH", ""), ARCH_PATH_LIST_SEP));
```

Comment:
We might want to cache results so we do not hit the filesystem too hard every time we expand a shader node graph in the outliner.